docs/proposals/instrumentation-of-services.md

```diff
@@ -0,0 +1,70 @@
+# Best practices for instrumenting infrastructure components in OpenShift
+
+All OpenShift components MUST expose the following endpoints on either their public HTTPS port, or a public HTTP port if none exists
+
+* `/healthz` - returns 200 and the text `ok` if the instance is alive
+* `/metrics` - returns a reasonable set of metrics in the Prometheus format for capturing the health and performance of the server in question
+
+Components MUST expose the following endpoints if they are exposed via a service or route
+
+* `/healthz/ready` - returns a 200 and the text `ok` if the instance is ready to begin receiving network traffic
+
+Components that only offer TCP endpoints should listen on a separate port and expose the appropriate checks above.
+
+Components that have equivalents to these endpoints already that are taken "as-is" from the upstream community MAY
+use those instead.
+
+Components that expose tenanted info that use authentication MUST use HTTPS to serve metrics - this is a general
+infrastructure requirement that all user info flow over TLS.
+
+
+## Metrics
+
+We use the Prometheus format to be consistent with the upstream Kubernetes community and because it is a natural fit
+for the Go ecosystem. COTS software that does not offer a native Prometheus endpoint should use an adapter to expose
+prometheus metrics, either in their process or as a sidecar container. Java components may use JMX, but we still
+recommend adapting to the prometheus format.
```

Comment:
allowing JMX w/o an adapter seems like it would fragment metrics (e.g. most metrics can be collect with Prometheus, except for the Java apps).  We should probably just put our foot down here too...